src/source/Dtls/Dtls.c

```diff
@@ -532,19 +603,21 @@ STATUS dtlsSessionPopulateKeyingMaterial(PDtlsSession pDtlsSession, PDtlsKeyingM
 }
 
 
-STATUS dtlsSessionGenerateLocalCertificateFingerprint(PDtlsSession pDtlsSession, PCHAR pBuff, UINT32 buffLen)
```

Comment:
Are we removing cert generation altogether?

Reply:
We are checking whether the app has specified a cert to use and if not, we will generate the cert, install the cert, extract the fingerprint and then delete the cert and the key once the cert is installed in the ephemeral cert store in the ssl context. The cert actually has reference counting so in reality we just relinquish our reference in case if we created one. The SSL session when terminated, will clear the cert reference. In case the cert is supplied by the application, the application should also free the reference. As we install the cert and extract the fingerprint during the create API call, the client application can release the reference to the cert and the key right after the create call.